webpack/assets/javascripts/react_app/components/CPUCoresInput/CPUCoresInput.js

```diff
@@ -0,0 +1,75 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { WarningTriangleIcon } from '@patternfly/react-icons';
+import { HelpBlock, FormGroup, Grid, Col, Row } from 'patternfly-react';
+import { translate as __ } from '../../common/I18n';
+import NumericInput from '../common/forms/NumericInput';
+import './cpuCoresInput.scss';
+import { noop } from '../../common/helpers';
+
+const CPUCoresInput = ({
+  label,
+  recommendedMaxValue,
+  maxValue,
+  defaultValue,
+  onChange,
+}) => {
+  const [value, setValue] = useState(defaultValue);
+
+  const getValidationState = () => {
+    if (value > recommendedMaxValue) return 'warning';
+    else if (value > maxValue) return 'error';
+    return null;
+  };
+
+  const handleChange = v => {
+    setValue(v);
+    onChange();
+  };
+
+  return (
+    <FormGroup validationState={getValidationState()}>
```

Comment:
The function is called twice (lines 31, 43), we can store the output before rendering once